lib/posix/options/eventfd.c

```diff
@@ -247,7 +247,9 @@ static int eventfd_ioctl_op(void *obj, unsigned int request, va_list args)
 			errno = EINVAL;
 			ret = -1;
 		} else {
-			efd->flags = flags;
+			int prev_flags = efd->flags & ~EFD_FLAGS_SET_INTERNAL;
+
+			efd->flags = flags | prev_flags;
```

Comment:
We could avoid extra variable by this:
```
efd->flags = (efd->flags & ~EFD_FLAGS_SET_INTERNAL ) | flags;
```
The line is still readable and not even that long.

Reply:
Thanks for your feedback. Personally I prefer the separate variable, but if that is an issue, of course I'll change it.